checker/jdk/nullness/src/org/w3c/dom/Document.java

```diff
@@ -0,0 +1,847 @@
+     * the implementation may update default attributes from other schemas.
+     * Applications should use <code>Document.normalizeDocument()</code> to
+     * guarantee these attributes are up-to-date.
+     * <br>When the node being renamed is an <code>Attr</code> that is
+     * attached to an <code>Element</code>, the node is first removed from
+     * the <code>Element</code> attributes map. Then, once renamed, either
+     * by modifying the existing node or creating a new one as described
+     * above, it is put back.
+     * <br>In addition,
+     * <ul>
+     * <li> a user data event <code>NODE_RENAMED</code> is fired,
+     * </li>
+     * <li>
+     * when the implementation supports the feature "MutationNameEvents",
+     * each mutation operation involved in this method fires the appropriate
+     * event, and in the end the event {
+     * <code>http://www.w3.org/2001/xml-events</code>,
+     * <code>DOMElementNameChanged</code>} or {
+     * <code>http://www.w3.org/2001/xml-events</code>,
+     * <code>DOMAttributeNameChanged</code>} is fired.
+     * </li>
+     * </ul>
+     * @param n The node to rename.
+     * @param namespaceURI The new namespace URI.
+     * @param qualifiedName The new qualified name.
+     * @return The renamed node. This is either the specified node or the new
+     *   node that was created to replace the specified node.
+     * @exception DOMException
+     *   NOT_SUPPORTED_ERR: Raised when the type of the specified node is
+     *   neither <code>ELEMENT_NODE</code> nor <code>ATTRIBUTE_NODE</code>,
+     *   or if the implementation does not support the renaming of the
+     *   document element.
+     *   <br>INVALID_CHARACTER_ERR: Raised if the new qualified name is not an
+     *   XML name according to the XML version in use specified in the
+     *   <code>Document.xmlVersion</code> attribute.
+     *   <br>WRONG_DOCUMENT_ERR: Raised when the specified node was created
+     *   from a different document than this document.
+     *   <br>NAMESPACE_ERR: Raised if the <code>qualifiedName</code> is a
+     *   malformed qualified name, if the <code>qualifiedName</code> has a
+     *   prefix and the <code>namespaceURI</code> is <code>null</code>, or
+     *   if the <code>qualifiedName</code> has a prefix that is "xml" and
+     *   the <code>namespaceURI</code> is different from "<a href='http://www.w3.org/XML/1998/namespace'>
+     *   http://www.w3.org/XML/1998/namespace</a>" [<a href='http://www.w3.org/TR/1999/REC-xml-names-19990114/'>XML Namespaces</a>]
+     *   . Also raised, when the node being renamed is an attribute, if the
+     *   <code>qualifiedName</code>, or its prefix, is "xmlns" and the
+     *   <code>namespaceURI</code> is different from "<a href='http://www.w3.org/2000/xmlns/'>http://www.w3.org/2000/xmlns/</a>".
+     * @since DOM Level 3
+     */
+    public Node renameNode(Node n,
+                           String namespaceURI,
```

Comment:
The Javadoc documentation indicates that `namespaceURI` is permitted to be `null`.
`namespaceURI` is also `@Nullable` in most other places in the pull request.  I suggest reviewing all uses of `namespaceURI` to see whether more should be `@Nullable`.  For example, despite the lack of documentation one way or the other, should it be `@Nullable` in `lookupPrefix` and in `isDefaultNamespace` elsewhere in this pull request?